src/Build/BuildCheck/API/ConfigurationContext.cs

```diff
@@ -9,22 +9,28 @@
 namespace Microsoft.Build.Experimental.BuildCheck;
 
 /// <summary>
-/// Holder of an optional configuration from .editorconfig file (not recognized by the infrastructure)
+/// Holder of configuration from .editorconfig file (not recognized by the infrastructure)
 /// </summary>
 public class ConfigurationContext
 {
-    private ConfigurationContext(CustomConfigurationData[] customConfigurationData)
+    private ConfigurationContext(CustomConfigurationData[] customConfigurationData, BuildAnalyzerConfiguration[] buildAnalyzerConfig)
     {
         CustomConfigurationData = customConfigurationData;
+        BuildAnalyzerConfig = buildAnalyzerConfig;
     }
 
-    internal static ConfigurationContext FromDataEnumeration(CustomConfigurationData[] customConfigurationData)
+    internal static ConfigurationContext FromDataEnumeration(CustomConfigurationData[] customConfigurationData, BuildAnalyzerConfiguration[] buildAnalyzerConfig)
```

Comment:
Thinking if we should cover the change by the e2e tests? Since we already have the custom buildcheck test, there is a chance to cover this case as well? What do you think?

Reply:
I suuport Farhad's idea.
Feel free to contact me if you have any questions about the e2e tests.

Reply:
These tests are currently disabled. I still think we should add tests to cover this functionality, so I created an issue to unblock this PR: https://github.com/dotnet/msbuild/issues/10291